### ISSUE TITLE
[video_player][iOS]: aspect ratio of m3u8 videos is different from .mp4 videos

### ISSUE THREAD

**Issue by IlyaMax** [flutter/flutter#97206]
tried on latest version of video_player 2.2.15 and this https://multiplatform-f.akamaihd.net/i/multi/will/bunny/big_buck_bunny_,640x360_400,640x360_700,640x360_1000,950x540_1500,.f4v.csmil/master.m3u8. 
**expected**: aspect ratio is not 1.0 
**actual**: aspect ratio equals to 1.0

**Comment by darshankawar** [comment]
@IlyaMax 
I think the link you shared got mixed up and it downloads as a single link which I presume are different videos. Can you provide proper links ?
Are you saying that the aspect ratio of these videos should not be same and should match native iOS behavior ?

**Comment by IlyaMax** [comment]
@darshankawar Sorry, I don't really understand what you mean. To reproduce you just need to pass the url to `VideoPlayerController.network` in example. On iOS I get wrong aspect ratio which is 1.0, but on android it is correct for same url.

**Comment by darshankawar** [comment]
@IlyaMax 
Thanks, yes. Using the video link you provided, I do see it shows 1.0. Does it happen only with this video link or with all videos ? ie, only with .m3u8 extension, as you said in the title some video can't get right aspect ratio ?

**Comment by IlyaMax** [comment]
@darshankawar using mp4 aspect ratio is correct. You can validate this with default mp4 bee video. I've reproduced this only for m3u8 videos but I'm not sure about other formats so titled "some videos"

**Comment by darshankawar** [comment]
Thanks for the update. I verified using .mp4 with which the aspect ratio (width and height) is as below:


https://user-images.githubusercontent.com/67046386/151298227-d0dcccd6-5164-4d11-9a05-93fa1e69f5e7.mov


Using a .m3u8 file, the aspect ratio is different, as below:


https://user-images.githubusercontent.com/67046386/151298408-7f668e11-67e0-4abb-b3a2-6ecdc5f30260.mov




`video_player: ^2.2.16`

Code sample used: https://pub.dev/packages/video_player/example

<details>
<summary> master, stable flutter doctor -v </summary>

```

[✓] Flutter (Channel master, 2.10.0-1.0.pre.307, on Mac OS X 10.15.4 19E2269
    darwin-x64, locale en-GB)
    • Flutter version 2.10.0-1.0.pre.307 at
      /Users/dhs/documents/fluttersdk/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision b7424c619c (3 hours ago), 2022-01-26 17:25:45 -0800
    • Engine revision c248c6e38e
    • Dart version 2.17.0 (build 2.17.0-63.0.dev)
    • DevTools version 2.10.0-dev.1

[!] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! Flutter recommends a minimum Xcode version of 13.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • Mac OS X 10.15.4 19E2269
      darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 97.0.4692.99

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.

[✓] Flutter (Channel stable, 2.8.1, on Mac OS X 10.15.4 19E2269 darwin-x64,
    locale en-GB)
    • Flutter version 2.8.1 at /Users/dhs/documents/fluttersdk/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 77d935af4d (13 hours ago), 2021-12-16 08:37:33 -0800
    • Engine revision 890a5fca2e
    • Dart version 2.15.1

[✓] Android toolchain - develop for Android devices (Android SDK version 30)
    • Android SDK at /Users/dhs/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Users/dhs/Library/Android/sdk
    • Java binary at: /Users/dhs/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/202.7486908/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.    

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Users/dhs/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/202.7486908/Android
      Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)        

[✓] VS Code (version 1.57.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (4 available)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1 18D61
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 93.0.4577.82


• No issues found!

```

</details>

**Comment by djsjr** [comment]
This aspectRatio = 1 occurs with my .m3u8 video links as well — same exact situation and behavior as OP.

**Comment by edurossi** [comment]
I had the same problem opened the issue #97478 . Now I close it and continue here. 

This is what I found:

I tried to find the problem and it started on version video_player-v2.2.12
The problem is in the file FLTVideoPlayerPlugin in setupEventSinkIfReadyToPlay
```
if (hasVideoTracks && height == CGSizeZero.height && width == CGSizeZero.width) {
    return;
}
```
if the 'hasVideoTracks' is removed (as it was on version 2.2.11), the video looks fine and you can get the right width and height. If not, you always get zero in width and height

A temporal workaround is to install the plugin as:
`video_player: 2.2.11`

**Comment by jmagman** [comment]
@AlexV525 More fallout from https://github.com/flutter/plugins/pull/4639

**Comment by jmagman** [comment]
HLS videos evidently don't have `tracks` https://stackoverflow.com/questions/52409687/cant-able-to-get-video-tracks-from-avurlasset-for-hls-videos-m3u8-format-for

**Comment by jmagman** [comment]
Adding a HLS test file: https://github.com/flutter/assets-for-api-docs/pull/178

**Comment by AlexV525** [comment]
Would https://stackoverflow.com/a/63843659 improve this?

**Comment by jmagman** [comment]
> Would https://stackoverflow.com/a/63843659 improve this?

That's what removing `hasVideoTracks` does, it only checks height and width.
I have a fix for this, I just want the test file at https://github.com/flutter/assets-for-api-docs/pull/178 first.

**Comment by rahul-insight** [comment]
We are also facing the same issue. Is there a solution planned for this?

**Comment by jmagman** [comment]
> We are also facing the same issue. Is there a solution planned for this?

Fix is out for review https://github.com/flutter/plugins/pull/4727

**Comment by jmagman** [comment]
Available in `video_player_avfoundation: 2.2.18`

**Comment by dgilperez** [comment]
Thanks a lot @jmagman, great speed! We will upgrade and report back in case this is not fixed for us.
